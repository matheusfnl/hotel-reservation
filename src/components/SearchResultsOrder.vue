<template>
  <span>
    Organizar por

    <span class="text-primary text-bold text-italic cursor-pointer" @click="showMenu = true">
      {{ selectedOption }}
      <q-icon name="expand_more" class="q-mr-sm" />

      <q-menu>
        <q-list style="min-width: 200px" v-for="item in menuOptions" :key="item">
          <q-item dense clickable v-close-popup @click="selectOption(item)">
            <q-item-section>{{ item }}</q-item-section>
          </q-item>
        </q-list>
      </q-menu>
    </span>
  </span>
</template>

<script lang="ts" setup>
import { ref } from 'vue'

const menuOptions = ['recommended', 'high_price', 'low_price']

const showMenu = ref(false)
const selectedOption = ref('recommended')

const selectOption = (option: string) => {
  selectedOption.value = option
}
</script>

<style scoped>
.inline-select :deep(.q-btn) {
  color: #1976d2;
}
</style>
