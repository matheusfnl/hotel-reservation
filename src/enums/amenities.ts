export enum Amenities {
  WI_FI = 'wifi',
  RESTAURANT = 'restaurant',
  ROOM_SERVICE = 'room_service',
  MEETING_ROOM = 'meeting_room',
  LAUNDRY = 'local_laundry_service',
  BREAKFAST = 'bakery_dining',
  PARKING = 'local_parking',
  FITNESS_CENTER = 'fitness_center',
  POOL = 'pool',
  SPA = 'spa',
  STEAM_ROOM = 'hot_tub',
  PETS = 'pets',
  PUB = 'sports_bar',
  AIR_CONDITIONING = 'ac_unit',
  SAFE = 'lock',
}
