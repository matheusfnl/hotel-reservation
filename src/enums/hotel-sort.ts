export enum Sort {
  RECOMMENDED = 'recommended',
  BEST_RATED = 'best_rated',
}
