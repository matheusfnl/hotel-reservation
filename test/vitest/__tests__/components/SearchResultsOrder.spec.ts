import { describe, it, expect } from 'vitest'
import { QMenu, QItem } from 'quasar'
import { mount } from '@vue/test-utils'

import SearchResultsOrder from '@/components/SearchResultsOrder.vue'

import { Sort } from '@/enums/hotel-sort'

const mountWrapper = (props: { modelValue: string }) => {
  return mount(SearchResultsOrder, {
    props,
  })
}

describe('SearchResultsOrder', () => {
  it('should render correctly with initial value', () => {
    const wrapper = mountWrapper({ modelValue: Sort.RECOMMENDED })
    const menu = wrapper.findComponent(QMenu)
    const items = menu.findAllComponents(QItem)

    expect(wrapper.html()).toMatchSnapshot()
    expect(items.length).toBe(0)
  })

  it('should show menu options when clicked', async () => {
    const wrapper = mountWrapper({ modelValue: Sort.RECOMMENDED })
    const trigger = wrapper.find('.cursor-pointer')

    await trigger.trigger('click')
    const menu = wrapper.findComponent(QMenu)
    const items = menu.findAllComponents(QItem)

    expect(items[0]!.text()).toContain(wrapper.vm.$t(`hotel.filters.options.${Sort.RECOMMENDED}`))
    expect(items[1]!.text()).toContain(wrapper.vm.$t(`hotel.filters.options.${Sort.BEST_RATED}`))
  })

  it('should emit update event when selecting an option', async () => {
    const wrapper = mountWrapper({ modelValue: Sort.RECOMMENDED })
    const trigger = wrapper.find('.cursor-pointer')

    await trigger.trigger('click')
    const menu = wrapper.findComponent(QMenu)
    const items = menu.findAllComponents(QItem)

    await items[1]!.trigger('click')

    expect(wrapper.emitted('update:modelValue')?.[0]).toEqual([Sort.BEST_RATED])
  })
})
