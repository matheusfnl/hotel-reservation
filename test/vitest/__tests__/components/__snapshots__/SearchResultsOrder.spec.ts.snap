// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchResultsOrder > should render correctly with initial value 1`] = `"<span>Organizar por <span class="text-primary text-bold text-italic cursor-pointer">Recomendados <i class="q-icon notranslate material-icons q-mr-sm" aria-hidden="true" role="presentation">expand_more</i><!----></span></span>"`;
